src/Game/Game.cpp

```diff
@@ -174,11 +174,11 @@ void Game::run()
         think();
         render();
         SDL_Delay(1);
-        while (!_statesForDelete.empty())
+        for (auto state : _statesForDelete)
         {
-            delete _statesForDelete.back();
-            _statesForDelete.pop_back();
+            delete state;
         }
+        _statesForDelete.clear();
```

Comment:
Am I missing something here? _statesForDelete is not a local variable, it's a class member, it will only be cleared when Game is destroyed. But states need to be deleted at the and of every frame. If I don't clear it, it will try to delete object by pointer which was already deleted in previous frame and crash.